crates/walrus-service/src/node/node_recovery.rs

```diff
@@ -18,20 +18,29 @@ pub struct NodeRecoveryHandler {
 
     // There can be at most one background shard removal task at a time.
     task_handle: Arc<Mutex<Option<tokio::task::JoinHandle<()>>>>,
+
+    // Configuration for node recovery.
+    config: NodeRecoveryConfig,
 }
 
 impl NodeRecoveryHandler {
-    pub fn new(node: Arc<StorageNodeInner>, blob_sync_handler: Arc<BlobSyncHandler>) -> Self {
+    pub fn new(
+        node: Arc<StorageNodeInner>,
+        blob_sync_handler: Arc<BlobSyncHandler>,
+        config: NodeRecoveryConfig,
+    ) -> Self {
         Self {
             node,
             blob_sync_handler,
             task_handle: Arc::new(Mutex::new(None)),
+            config,
         }
     }
 
     /// Starts the node recovery process to recover blobs that are certified before the given epoch.
     /// For blobs that are certified after `certified_before_epoch`, the event processing is in
     /// charge of making sure the blob is stored at all shards.
+    // TODO(WAL-864): Refactor this function to make it readable.
```

Comment:
This function is getting long and I want to refactor it, but preferably in a separate PR since I want this PR to be easier to review by looking at the diff.